src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java

```diff
@@ -0,0 +1,172 @@
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+import static org.opensearch.security.support.ConfigConstants.OBO_DEFAULT_EXPIRY_SECONDS;
+import static org.opensearch.security.support.ConfigConstants.OBO_MAX_EXPIRY_SECONDS;
+
+public class CreateOnBehalfOfTokenAction extends BaseRestHandler {
+
+    private static final List<Route> routes = addRoutesPrefix(
+        ImmutableList.of(new NamedRoute.Builder().method(POST).path("/generateobotoken").uniqueName("security:obo/create").build()),
+        "/_plugins/_security/api"
+    );
+
+    private JwtVendor vendor;
+    private final ThreadPool threadPool;
+    private final ClusterService clusterService;
+
+    private ConfigModel configModel;
+
+    private DynamicConfigModel dcm;
+
+    @Subscribe
+    public void onConfigModelChanged(ConfigModel configModel) {
+        this.configModel = configModel;
+    }
+
+    @Subscribe
+    public void onDynamicConfigModelChanged(DynamicConfigModel dcm) {
+        this.dcm = dcm;
+
+        Settings settings = dcm.getDynamicOnBehalfOfSettings();
+
+        Boolean enabled = Boolean.parseBoolean(settings.get("enabled"));
+        String signingKey = settings.get("signing_key");
+        String encryptionKey = settings.get("encryption_key");
+
+        if (!Boolean.FALSE.equals(enabled) && signingKey != null && encryptionKey != null) {
+            this.vendor = new JwtVendor(settings, Optional.empty());
+        } else {
+            this.vendor = null;
+        }
+    }
+
+    public CreateOnBehalfOfTokenAction(final Settings settings, final ThreadPool threadPool, final ClusterService clusterService) {
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public String getName() {
+        return getClass().getSimpleName();
+    }
+
+    @Override
+    public List<Route> routes() {
+        return routes;
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        switch (request.method()) {
+            case POST:
+                return handlePost(request, client);
+            default:
+                throw new IllegalArgumentException(request.method() + " not supported");
+        }
+    }
+
+    private RestChannelConsumer handlePost(RestRequest request, NodeClient client) throws IOException {
+        return new RestChannelConsumer() {
+            @Override
+            public void accept(RestChannel channel) throws Exception {
+                final XContentBuilder builder = channel.newBuilder();
+                BytesRestResponse response;
+                try {
+                    if (vendor == null) {
+                        channel.sendResponse(
+                            new BytesRestResponse(
+                                RestStatus.SERVICE_UNAVAILABLE,
+                                "on_behalf_of is either disabled or the configuration is invalid"
```

Comment:
This message should help an end user understand what action to take, `The OnBehalfOf token generating API has been disabled, see ...{link to doc}... for more information on this feature.`

Note; it should be using full words and legit grammar since its visible by end users.

Reply:
Good call. Should we capture this in a separate documentation related issue? We can update this line once the documentation website has the updates, since we don't have an available link for now.

Reply:
> Should we capture this in a separate documentation related issue

IMO Its a bug to have unactionable error messages, but we can handle that outside of the core feature of this pull request if you'd like to make an issue.

Reply:
Sounds good, I have re-edit this msg, and leave the documentation link blanked for now since we don't have an available link. I will capture this as a follow-up issue for adding the link once it is ready.